solr/solrj/src/java/org/apache/solr/common/params/CollectionAdminParams.java

```diff
@@ -80,6 +80,17 @@
    */
   String COLL_CONF = "collection.configName";
 
+  /**
+   * The name of the collection with which a collection is to be co-located
```

Comment:
It would be helpful to explicit here what this really means and assumes.

Reply:
Something along "Collection A with the attribute withCollection collection B implies two things: Collection B has only one shard, and on every node where collection A has a replica, there will also be a replica of collection B's shard".

Reply:
And likely need to specify (here or elsewhere) that this is done for collection creation and adding replicas for Collection A, but not moving them...

Reply:
At this stage we're not fully sure yet how far this feature will go in ensuring this constraint, and where it will be implemented... This flag here just for indicating the constraint, implementation notes should go to the plugins that support this flag.

Reply:
I think it's valid to specify and document what this flag does by default. Obviously plugins should add documentation on additional logic that they use for these flags as well.